### ISSUE TITLE
Transfer Learning

### ISSUE THREAD

**Issue by GianniCristian** [fizyr/keras-retinanet#280]
Dear All,

I would like to ask you if there is the transfer learning option. I've seen there is the option for training on a custom dataset, but I assume it has done from scratch, right?

Thanks

**Comment by awilliamson** [comment]
The backbone is a direct application of Transfer Learning, where you can load in imagenet weights, and then optionally freeze layers, and fine-tune the upper ones. If you're referring to transfering the entire retinanet. That should still be possible by taking a pretrained RetinaNet application on VOC, and trying to transfer it to a similar 'everyday' object domain?

**Comment by Knight13** [comment]
Is there any available pre-trained RetinaNet model on VOC?

**Comment by awilliamson** [comment]
Not to my knowledge. There only exists a COCO dataset. And currently, with development happening, some weight files may become outdated depending on changes. It shouldn't take too long to generate good accuracy models for VOC. Is there a particular reason you wish to Transfer Learn utilising this entire model?

**Comment by GianniCristian** [comment]
Thanks for the prompt responses! I would like to ask you another dummy one. If I've understood correctly, the command below train the network on a custom dataset by using imagenet weights:

`# Using the installed script:
retinanet-train csv /path/to/csv/file/containing/annotations /path/to/csv/file/containing/classes`

If I wanna train from scratch, I should do:

`retinanet-train --weights None csv <annotation.csv> <classmapping.csv>`

Thanks

**Comment by de-vri-es** [comment]
Currently that wouldn't work due to an oversight on our part. Please take a look at #275 for training from scratch.

**Comment by hgaiser** [comment]
Indeed, as mentioned, the `--weights` argument can be used for transfer learning. Personally, even if your domain is very different from imagenet, I would always start with imagenet as initialization. In my opinion it can't be worse than completely random weights. You need a significant amount of data to train from scratch..

Assuming the original issue is resolved, I will close this. If you face additional issues, feel free to create a new one.
